modules/storage-api/src/main/java/org/apache/ignite/internal/storage/util/StorageUtils.java

```diff
@@ -129,6 +131,8 @@ public static void throwExceptionDependingOnStorageState(StorageState state, Str
                 throw new StorageRebalanceException(createStorageInProcessOfRebalanceErrorMessage(storageInfo));
             case CLEANUP:
                 throw new StorageException(createStorageInProcessOfCleanupErrorMessage(storageInfo));
+            case DESTROYING:
+                throw new StorageException(createStorageDestroyedErrorMessage(storageInfo));
```

Comment:
I propose to add a special exception so that we can handle the situation that the index is destroyed and we simply skip it.

Reply:
I think we can reuse the `StorageClosedException` for now and change it to a more specific type later, if needed